tools/README.md

```diff
@@ -1,64 +1,65 @@
-# Tools for arti-cache
+# Tools for generating custom directory information
 
-## Overview
+To reduce bandwidth use of lightarti-rest, the library relies on custom Tor directory information. The scripts in the `tools/` directory generate the required files, that can then be downloaded by apps for use with lightarti-rest.
 
-To send some data anonymously over the network, Lightarti-rest relies on a
-subset of relays of the Tor network.
+The scripts in this directory aim to pick reliable nodes so that ideally the consensus can be used for up to a week. We refer to the [accompanying Jupyter notebook](churn_analysis.ipynb) for a historical analysis of churn on the Tor network that shows shows that this approach is viable.
 
-This script allows to set up a snapshot of the most reliable Tor relays
-for faster circuit setup.
+The tool provides three functions:
 
-You will have to create a custom directory authority, and to periodically
-run this script on a trusted server.
+1. Setting up a custom directory authority. We will use the keys of this authority to sign the smaller custom consensus files. This step only needs to be run once.
+2. Generating a new custom consensus. Based on the latest consensus information, the script produces a smaller consensus of reliable nodes.  A new custom consensus should ideally be generated at least once a day. Generating a new custom consensus requires the keys from the previous step.
+3. Generating a churn file. As a custom consensus ages, some nodes may no longer be available. To increase efficiency, lightarti-rest can take as input a very small churn file that lists available nodes. A churn file is generated with respect to a specific custom consensus. Ideally it is updated every hour.
 
-Then the app will have to download the files, put them in a directory,
-and send the path to this directory to the library.
+We describe these in more detail below.
```

Comment:
Can you add an example of size of the consensus and the churn file? I think it's 1MB and 10kB. This is interesting for readers.

Reply:
Good point, I am going to add some reasonable measurements. Essentially something like this:

The size of a consensus retrieved from the tor authorities in microconsensus format is about 2.2 MB.

The size of a shortened consensus will depend on the number of relay we decide to include, if we keep the default values (120 routers), it will have a size of about 45 KB.

The size of the churn file will increase over time, as more relays become unreachable. Each unreachable relay increase the size of the churn file by 41 bytes (40 bytes for the fingerprint of the relay plus 1 byte for the new line). So typically, its size will be less 1 KB.

Reply:
That should go in the README file...